src/masternode.cpp

```diff
@@ -704,7 +704,11 @@ bool CMasternodePing::CheckAndUpdate(int& nDos, bool fRequireEnabled, bool fChec
     LogPrint(BCLog::MNPING, "%s: New Ping - %s - %s - %lli\n", __func__, GetHash().ToString(), blockHash.ToString(), sigTime);
 
     if (isMasternodeFound && pmn->protocolVersion >= ActiveProtocol()) {
-        if (fRequireEnabled && !pmn->IsEnabled()) return false;
+
+        // Update ping only if the masternode is active/enabled
+        if (fRequireEnabled && (!pmn->IsEnabled() && !pmn->IsInActiveState())) {
```

Comment:
"IsInActiveState" is a bit weird name imo (considering that we are checking something called itself "active state").
Since we already have `IsEnabled()`, maybe would be more consistent to call this other one `IsPreEnabled()`? 
Or, alternatively, you could directly define a single function (e.g. `IsActive()`) that includes both active states in OR, similar to the one we already  have in the GUI mnmodel:
```cpp
bool MNModel::isMNActive(QString mnAlias)
{
    int activeState = getMNState(mnAlias);
    return activeState == CMasternode::MASTERNODE_PRE_ENABLED || activeState == CMasternode::MASTERNODE_ENABLED;
}
```
And use `!pmn->IsActive()` here.

Reply:
yeah, I thought about it as well but the "active state" concept is already being used for pre-enabled state only (look at `CMasternode::Status()`).

Will push the `IsPreEnabled` method and moving forward, we can standardize the concepts properly.